client/components/Recipe.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import { useSelector } from 'react-redux'
+
+export default function Recipe() {
+  const { id } = useParams()
+  const recipes = useSelector((state) => state.recipes)
+  const recipe = recipes[id]
+
+  const label = recipe.recipe.label
+  const largeImage = recipe.recipe.images.LARGE?.url
+  const regularImage = recipe.recipe.images.REGULAR.url
+  const ingredientsArr = recipe.recipe.ingredients
+
+  const dietaryArr = recipe.recipe.healthLabels
+  const selectiveDietary = dietaryArr.filter((word) => {
+    return word === 'Vegan' || word === 'Vegetarian' || word === 'Gluten-Free'
+  })
+
+  return (
```

Comment:
Since all of this is dependant on the stuff about I would have a loading state that waits for all the data you need to be finished.
Just but an if statement above it

```javascript
  // Have all of the stuff you need in here
  if(){
     return '...loading'
  }
  
   return (
      // everything in your current return
   )
```

Reply:
The Recipe component is displayed on a separate route. Prior to clicking on each individual recipe to take us to that route, all their information has already been loaded to the global store, from which Recipe component can access to extract the needed information.

Reply:
What happens if someone goes directly to this page?

Reply:
Oh I see whats happening, Your just pulling the 1 recipe from your global store which is all your recipes. I would still say there should be an api call for the single recipe here. This goes back to my comment about the api getting all of your recipes in your database. if someone just comes to this page directly (manually types it into the url) then they would be making an api call to get all of the recipes even though they only want/need to see this one. It is fine if there is only small amount of recipes but once you start to get into the hundreds then thats a lot of unnecessary data being sent to the app

Reply:
That makes more sense now. We understand the problem but might not be able to tackle this problem at this stage. Thanks for pointing this out to us.